src/vulkan/vulkanutils.cpp

```diff
@@ -3,9 +3,140 @@
 #include <iomanip>
 #include <sstream>
 
-// TODO: fix namespaces
+#if defined(__APPLE__) || defined(__linux)
+#include <dlfcn.h>
+#endif
+
 namespace dw::vulkan {
 
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+VulkanRTLPtr GetRuntimeLibs()
+{
+    VulkanRTLPtr vulkanLibrary;
+
+#if defined (_WIN32)
+    vulkanLibrary = LoadLibrary("vulkan-1.dll");
+#elif defined (__linux )
+    vulkanLibrary = dlopen("libvulkan.so.1", RTLD_NOW | RTLD_LOCAL);
+#elif defined (__APPLE__ )
+    vulkanLibrary = dlopen("libvulkan.1.dylib", RTLD_NOW | RTLD_LOCAL);
+#else
+    std::cerr << "The DireWolf renderer is not yet setup for Vulkan on this platform. Supported operating systems are Linux, Windows and Mac" << std::endl;
```

Comment:
This should never be reached, as we check this in the header. `assert(false && "error message");`

Reply:
Yeah, but I'm still conflicted if the header is the best place to put those things.. maybe we should break it earlier from outside in DireWolf as you said? I just think it should hold as a stand alone module as well.